src/encode/pattern/mod.rs

```diff
@@ -1031,6 +1083,46 @@ mod tests {
         assert_eq!(buf, b"missing value");
     }
 
+    #[test]
+    #[cfg(all(feature = "simple_writer", feature = "log_kv"))]
+    fn kv() {
```

Comment:
tests look good, can you also check some of the error cases? Take a look at #359 if you need some help

Reply:
I adapted the `test_from_piece_to_chunk` test from your PR, thanks!